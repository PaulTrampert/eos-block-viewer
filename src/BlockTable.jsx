import React from 'react';
import PropTypes from 'prop-types';
import Table from 'react-bootstrap/Table';
import BlockRow from './BlockRow';
import EmptyTableView from './EmptyTableView';
import TableLoadingView from './TableLoadingView';
import styles from './BlockTable.module.css';

class BlockTable extends React.Component {
  render = () => {
    let {
      blocks,
      loadingBlocks
    } = this.props;

    return (
      <Table striped hover className={styles.table}>
        <thead>
          <tr>
            <th className={styles.hash}>Block Hash</th>
            <th className={styles.timestamp}>Timestamp</th>
            <th className={styles.actions}># Actions</th>
          </tr>
        </thead>
        <tbody>
          {
            loadingBlocks && <TableLoadingView />
          }
          {
            !loadingBlocks && !!blocks.length && blocks.map(block => (
              <BlockRow key={block.id} block={block} />
            ))
          }
          {
            !loadingBlocks && !blocks.length && <EmptyTableView />
          }
        </tbody>
      </Table>
    );
  }
}

BlockTable.propTypes = {
  blocks: PropTypes.array
};

export default BlockTable;