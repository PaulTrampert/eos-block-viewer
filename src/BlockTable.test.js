import React from 'react';
import {shallow} from 'enzyme';
import BlockTable from './BlockTable';

describe('BlockTable', () => {
  let subject;
  let blocks;

  beforeEach(() => {
    blocks = [
      {
        id: '1',
        timestamp: '2020-02-05T03:46:32.000',
        transactions: []
      },
      {
        id: '2',
        timestamp: '2020-02-04T03:46:32.000',
        transactions: []
      },
      {
        id: '3',
        timestamp: '2020-02-03T03:46:32.000',
        transactions: []
      },
    ];

    subject = shallow(<BlockTable blocks={blocks} />);
  });

  describe('render', () => {
    describe('when blocks are loading', () => {
      beforeEach(() => {
        subject.setProps({loadingBlocks: true});
      });

      it('renders the TableLoadingView', () => {
        expect(subject).toMatchSnapshot();
      });
    });

    describe("when blocks are loaded", () => {
      it('renders a table with all the blocks', () => {
        expect(subject).toMatchSnapshot();
      });
    });

    describe('when there are no blocks loaded', () => {

      beforeEach(() => {
        subject.setProps({blocks:[]});
      });

      it('renders the EmptyTableView', () => {
        expect(subject).toMatchSnapshot();
      });
    });
  });
});