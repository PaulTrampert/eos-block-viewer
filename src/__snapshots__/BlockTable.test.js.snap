// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockTable render when blocks are loaded renders a table with all the blocks 1`] = `
<ForwardRef
  hover={true}
  striped={true}
>
  <thead>
    <tr>
      <th>
        Block Hash
      </th>
      <th>
        Timestamp
      </th>
      <th>
        # Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <BlockRow
      block={
        Object {
          "id": "1",
          "timestamp": "2020-02-05T03:46:32.000",
          "transactions": Array [],
        }
      }
      key="1"
    />
    <BlockRow
      block={
        Object {
          "id": "2",
          "timestamp": "2020-02-04T03:46:32.000",
          "transactions": Array [],
        }
      }
      key="2"
    />
    <BlockRow
      block={
        Object {
          "id": "3",
          "timestamp": "2020-02-03T03:46:32.000",
          "transactions": Array [],
        }
      }
      key="3"
    />
  </tbody>
</ForwardRef>
`;

exports[`BlockTable render when blocks are loading renders the TableLoadingView 1`] = `
<ForwardRef
  hover={true}
  striped={true}
>
  <thead>
    <tr>
      <th>
        Block Hash
      </th>
      <th>
        Timestamp
      </th>
      <th>
        # Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <TableLoadingView />
  </tbody>
</ForwardRef>
`;

exports[`BlockTable render when there are no blocks loaded renders the EmptyTableView 1`] = `
<ForwardRef
  hover={true}
  striped={true}
>
  <thead>
    <tr>
      <th>
        Block Hash
      </th>
      <th>
        Timestamp
      </th>
      <th>
        # Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <EmptyTableView />
  </tbody>
</ForwardRef>
`;
